### ISSUE TITLE
508-defect-1 ❗ Launchblocker [COGNITION]: Back to Top Component - buttons must have discernible text

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#18423]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a screen reader user, I want to understand the button's purpose and not hear extra information like the Font Awesome up arrow being announced. As a mobile screen reader user, I want to hear an accessible label that announces "Back to top, button".

## Details

The Back to Top button is using CSS `::before` and `::after` attributes to add text. It would be better to use spans so screen readers ignore the Font Awesome arrow and always have an accessible label available, regardless of size. Code snippets below.

## Acceptance Criteria

- [ ] Storybook axe violation about button not having discernible text is no longer present
- [ ] Screen readers announce an accessible label
- [ ] Visual display is maintained

## Proposed Solution (if known)

```diff
<button type="button" class="usa-button va-top-button-transition-in">
+ <span aria-hidden="true">
+   <i>Font awesome arrow</i>
+ </span>
+ <span class="UNIQUE_CLASS">Back to top</span>
</button>
```

```css
@media screen and (max-width: 768px) {
  span.UNIQUE_CLASS {
    /* Use the .vads-u-visibility--screen-reader to make text SR-only for mobile */
    /* https://design.va.gov/utilities/visibility */
    border: 0;
    clip: rect(0,0,0,0);
    -webkit-clip-path: inset(50%);
    clip-path: inset(50%);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute !important;
    width: 1px;
    word-wrap: normal !important}
  }
}
```

## WCAG or Vendor Guidance (optional)

* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

**Comment by bkjohnson** [comment]
I think this satisfies the AC: https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/562/files/0af6822e42842e90a480df27a403ebbb01cc604d..0d0219475d741f2de1ec21057ba6b3a2b3f6c260

I added the duplicate `<span>` for "Back to top" in with the icon because of how `vads-u-visibility--screen-reader` and the responsive prefixes work. I would have had to use JS to add that _only_ when the screen size is below a certain size and remove it otherwise. I couldn't just have that be the default and tack on another class like `medium-screen:vads-u-visibility-visible`.

**Comment by 1Copenut** [comment]
@bkjohnson I took a look at the linked branch and ran it through a couple of screen readers locally. The HTML structure announces the "Back to top" text twice because we're only hiding the first text span visually. Could we streamline it to one text span and override how it's displayed (or not) using CSS only? Maybe something like:

```css
/* Visually override the SR-only styles at tablet or wider */
@media screen and (min-width: 768px) #back-to-top-container button span {
    clip: auto;
    -webkit-clip-path: initial;
    display: inline-block;
    height: auto;
    margin: 0;
    overflow: auto;
    position: relative!important;
    width: auto;
}
```

```css
/* Text is visually hidden by default */
#back-to-top-container button span {
    clip: rect(0,0,0,0);
    -webkit-clip-path: inset(50%);
    clip-path: inset(50%);
    display: none;
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute!important;
    width: 1px;
    word-wrap: normal!important;
}
```

**Comment by bkjohnson** [comment]
Ok, I've [updated the branch](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/562/commits/af23320c4006d279dad5e28c906f6e9aa4cc1bed) to follow the original recommendation. How is this?

**Comment by bkjohnson** [comment]
Fixed in https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/562 according to a [Slack thread](https://dsva.slack.com/archives/C01DBGX4P45/p1611008184002800?thread_ts=1610125544.020700&cid=C01DBGX4P45)
